src/cascadia/TerminalSettingsModel/ApplicationState.cpp

```diff
@@ -0,0 +1,158 @@
+        newState->_load();
+        s = newState;
+        return *newState;
+    }
+
+    // Method Description:
+    // - Deserializes a state.json document into an ApplicationState.
+    // - *ANY* errors will result in the creation of a new empty state
+    void ApplicationState::_load()
+    {
+        wil::unique_hfile hFile{ CreateFileW(_path.c_str(),
+                                             GENERIC_READ,
+                                             FILE_SHARE_READ | FILE_SHARE_WRITE,
+                                             nullptr,
+                                             OPEN_EXISTING,
+                                             FILE_ATTRIBUTE_NORMAL,
+                                             nullptr) };
+
+        if (hFile)
+        {
+            try
+            {
+                const auto data{ ReadUTF8TextFileFull(hFile.get()) };
+
+                std::string errs; // This string will receive any error text from failing to parse.
+                std::unique_ptr<Json::CharReader> reader{ Json::CharReaderBuilder::CharReaderBuilder().newCharReader() };
+
+                // Parse the json data into either our defaults or user settings. We'll keep
+                // these original json values around for later, in case we need to parse
+                // their raw contents again.
+                Json::Value root;
+                // `parse` will return false if it fails.
+                if (reader->parse(data.data(), data.data() + data.size(), &root, &errs))
+                {
+                    LayerJson(root);
+                }
+            }
+            CATCH_LOG();
+        }
+    }
+
+    // Method Description:
+    // - Deserializes a JSON document into the current ApplicationState
+    void ApplicationState::LayerJson(const Json::Value& document)
+    {
+        JsonUtils::GetValueForKey(document, CloseAllTabsWarningDismissedKey, _CloseAllTabsWarningDismissed);
+        JsonUtils::GetValueForKey(document, LargePasteWarningDismissedKey, _LargePasteWarningDismissed);
+        JsonUtils::GetValueForKey(document, MultiLinePasteWarningDismissedKey, _MultiLinePasteWarningDismissed);
+    }
+
+    // Method Description:
+    // - Creates a JSON document from the current ApplicationState
+    Json::Value ApplicationState::ToJson() const
+    {
+        Json::Value document{ Json::objectValue };
+        JsonUtils::SetValueForKey(document, CloseAllTabsWarningDismissedKey, _CloseAllTabsWarningDismissed);
+        JsonUtils::SetValueForKey(document, LargePasteWarningDismissedKey, _LargePasteWarningDismissed);
+        JsonUtils::SetValueForKey(document, MultiLinePasteWarningDismissedKey, _MultiLinePasteWarningDismissed);
+        return document;
+    }
+
+    // Method Description:
+    // - Unhooks the current application state from global storage so that a subsequent request will
+    //   reload it (public)
+    void ApplicationState::Reload()

```

Comment:
nit: thoughts on renaming this? Since its not actually doing another load, maybe we can just call it `invalidate`? So we avoid the confusion